cli_helpers/tabular_output/preprocessors.py

```diff
@@ -227,5 +231,5 @@ def _format_number(field, column_type):
             return format(field, float_format)
         return field
 
-    data = [[_format_number(v, column_types[i]) for i, v in enumerate(row)] for row in data]
+    data = ([_format_number(v, column_types[i]) for i, v in enumerate(row)] for row in data)
     return data, headers
```

Comment:
hm I don't know which line you're referring to 😄 

I've dropped the cast to `iter(data)` because I've added unit tests to enforce iterable compatibility...